pkg/policy/updaterequest.go

```diff
@@ -0,0 +1,160 @@
+package policy
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gardener/controller-manager-library/pkg/logger"
+	kyverno "github.com/kyverno/kyverno/api/kyverno/v1"
+	urkyverno "github.com/kyverno/kyverno/api/kyverno/v1beta1"
+	"github.com/kyverno/kyverno/pkg/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+func (pc *PolicyController) updateUR(policyKey string, policy kyverno.PolicyInterface) {
+	logger := pc.log.WithName("updateUR").WithName(policyKey)
+	logger.Info("update URs on policy event")
+
+	mutateURs := pc.listMutateURs(policyKey, nil)
+	generateURs := pc.listGenerateURs(policyKey, nil)
+	updateUR(pc.kyvernoClient, policy.GetName(), append(mutateURs, generateURs...), pc.log.WithName("updateUR"))
+
+	for _, rule := range policy.GetSpec().Rules {
+		var ruleType urkyverno.RequestType
+		if rule.IsMutateExisting() {
+			ruleType = urkyverno.Mutate
+		} else {
+			// TODO: assign generate ruleType
+			continue
+		}
+
+		triggers := getTriggers(rule)
+		for _, trigger := range triggers {
+			var urs []*urkyverno.UpdateRequest
+			if ruleType == urkyverno.Mutate {
+				urs = pc.listMutateURs(policyKey, trigger)
+			} else {
+				urs = pc.listGenerateURs(policyKey, trigger)
+			}
+
+			logger.V(4).Info("UR was created", "rule", rule.Name, "rule type", ruleType, "trigger", trigger.Namespace+trigger.Name)
+
+			if urs != nil {
+				continue
+			}
+
+			logger.Info("creating new UR")
+			ur := newUR(policy, trigger, ruleType)
+			new, err := pc.kyvernoClient.KyvernoV1beta1().UpdateRequests(config.KyvernoNamespace).Create(context.TODO(), ur, metav1.CreateOptions{})
+			if err != nil {
```

Comment:
similarly in case of generate , we are doing couple of more operations instead of creating UR

Reply:
What other operations?

Reply:
check the UR exists  based on the policy-name label

Reply:
It was done here:
https://github.com/kyverno/kyverno/pull/3717/files#diff-04ec1fa2727eec2b8a4712cb1307dc92165ac98a2f9adc476af43d973a89ab94R33-R45

Reply:
forgot to add , i am also trying to get the policycontext and applying the background checks